### ISSUE TITLE
Enabling editing an advertisement

### ISSUE THREAD

**Issue by aashimawadhwa** [PalisadoesFoundation/talawa-admin#1157]
**Is your feature request related to a problem? Please describe.**
We have ensured CRUD operations throughout the talawa( wherever necessary). We require an edit ad modal in the Advertisement section on all the created ads.

**Describe the solution you'd like**
An editing ad modal on all created ads, ensuring fields similar to create ads modal.

**Additional context**
NA

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by Vaishali054** [comment]
@aashimawadhwa can you assign this to me

**Comment by Cioppolo14** [comment]
@Vaishali054 I assigned you another issue, so you have reached 2 issues. 

Our policy is to assign no more than two issues to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.

**Comment by ibishal** [comment]
can you assign me this?

**Comment by Vaishali054** [comment]
@Cioppolo14 , I unassigned myself from one of the issues(#1136), it needs the issue with the pinned post(#1122) to be resolved in order to write test cases for the same.

**Comment by 01vishu** [comment]
@Cioppolo14 and @aashimawadhwa, I am interested in contributing to your project and would like to request to be assigned to the project. I believe I have the necessary skills and expertise to make meaningful contributions to the project.

**Comment by noman2002** [comment]
@01vishu @ibishal I am assigning this to @Vaishali054 As she commented first and unassigned another issue which is assigned to her. Please take other open issues.

**Comment by ShatilKhan** [comment]
Please assign me

**Comment by Vaishali054** [comment]
@ShatilKhan I am already working on this, just waiting for the api to incorporate the feature to update, you can checkout other open issues.

**Comment by Vaishali054** [comment]
@aashimawadhwa This issue has been resolved, could you please close it? Thank you!

**Comment by palisadoes** [comment]
@Vaishali054 

Unfortunately, this PR should not have been merged as we have discovered that:

1.  the tests were faulty and 
2. the the corresponding mutation in the API was not done.

So this feature would not work worked.

1. If this assessment is true, then please be more professional in your approach to our repositories. 
2. If it is not, then please let us know your view on the matter

- https://github.com/PalisadoesFoundation/talawa-admin/pull/1268#issuecomment-1868463538

**Comment by palisadoes** [comment]
The fastest solution is to revert the PR. Too many other PRs are dependent on this

**Comment by palisadoes** [comment]
1. Reopening so that this can be tackled again.
2. The original PR can be used as a guide
3. A corresponding mutation in the API will need to be created beforehand and an issue will need to be created for that by who ever is assigned this new issue

**Comment by aarishshahmohsin** [comment]
Can I be assigned this issue?

**Comment by Vaishali054** [comment]
@palisadoes I should have been more responsible regarding this. @SiddheshKukade was going to add the mutation to the talawaAPI as discussed in the PR comments.

**Comment by palisadoes** [comment]
@Vaishali054

1. We all make mistakes.
2. I'll reassign this to you. Please fix it.

**Comment by Vaishali054** [comment]
Thank you for the confidence @palisadoes.

**Comment by palisadoes** [comment]
1. The faster we recognize failure, seek assistance and escalate, the better it is for the community on the whole. 
2. Only better collaboration will improve Talawa.

**Comment by palisadoes** [comment]
@Vaishali054 Are you still working on this?

**Comment by Vaishali054** [comment]
We are fine tuning the backend part of this PR. I have made a PR for it. Once that is merged then I will update the PR for this issue with latest changes otherwise it won't pass the checks
